src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/ComickFun.kt

```diff
@@ -1,450 +1,323 @@
-            ?: fetchMangaDetails(manga).map { mangaIdCache[mangaSlug] }
+    override fun headersBuilder() = Headers.Builder().apply {
+        add("Referer", "$baseUrl/")
+        add("User-Agent", "Tachiyomi ${System.getProperty("http.agent")}")
     }
 
-    /** Returns an identifier referred to as `hid` for chapter **/
-    private fun hid(chapter: SChapter) = "$baseUrl${chapter.url}".toHttpUrl().pathSegments[2].substringBefore("-")
+    override val client: OkHttpClient = network.client.newBuilder().apply {
+        rateLimit(
+            4, 1, TimeUnit.SECONDS
+        )
+    }.build()
 
-    /** Returns an identifier referred to as a  `slug` for manga **/
-    private fun slug(manga: SManga) = "$baseUrl${manga.url}".toHttpUrl().pathSegments[1]
+    override fun getFilterList() = FilterList(
+        filters.getFilters()
+    )
 
     /** Popular Manga **/
-
-    override fun fetchPopularManga(page: Int) = fetchSearchManga(page, "", FilterList(emptyList()))
-    override fun popularMangaRequest(page: Int) = throw UnsupportedOperationException("Not used")
-    override fun popularMangaParse(response: Response) = throw UnsupportedOperationException("Not used")
-
-    /** Latest Manga **/
-    override fun latestUpdatesParse(response: Response): MangasPage {
-        val noResults = MangasPage(emptyList(), false)
-        if (response.code == 204)
-            return noResults
-        return json.decodeFromString<List<SManga>>(
-            deserializer = ListSerializer(deepSelectDeserializer("md_comics")),
-            response.body!!.string()
-        ).let { MangasPage(it, true) }
-    }
-
-    override fun latestUpdatesRequest(page: Int): Request {
-        val url = "$apiBase/chapter".toHttpUrl().newBuilder()
-            .addQueryParameter("page", "${page - 1}")
-            .addQueryParameter("device-memory", "8")
-        return GET("$url", headers)
-    }
-
-    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
-        if (!query.startsWith(SLUG_SEARCH_PREFIX))
-            return super.fetchSearchManga(page, query, filters)
-
-        // deeplinking
-        val potentialUrl = "/comic/${query.substringAfter(SLUG_SEARCH_PREFIX)}"
-        return fetchMangaDetails(SManga.create().apply { this.url = potentialUrl })
-            .map { MangasPage(listOf(it.apply { this.url = potentialUrl }), false) }
-            .onErrorReturn { MangasPage(emptyList(), false) }
+    override fun popularMangaRequest(page: Int): Request {
+        return client.newCall(
+            GET(
+                API_BASE.toHttpUrl().newBuilder().apply {
+                    addPathSegment("search")
+                    addQueryParameter("sort", "user_follow_count")
+                    addQueryParameter("page", "$page")
+                    addQueryParameter("tachiyomi", "true")
+                }.build().toString(),
+                headers, CacheControl.FORCE_NETWORK
```

Comment:
```suggestion
                headers
```
You don't need to force network unless the website's cache control is glitchy. Same for other places.

Reply:
Done